packages/block-library/src/block/edit.js

```diff
@@ -215,10 +220,18 @@ export default function ReusableBlockEdit( {
 		  } )
 		: {};
 
-	useEffect(
-		() => setBlockEditMode( setBlockEditingMode, innerBlocks ),
-		[ innerBlocks, setBlockEditingMode ]
-	);
+	// Sync the editing mode of the pattern block with the inner blocks.
+	useEffect( () => {
+		const currentInnerBlocks = registry
```

Comment:
Get the `innerBlocks` from the registry so that we don't have to put it in the dependency. That is, we don't have to reset the editing mode for every inner block whenever something changes as a small performance improvement. Since we shouldn't be able to add or remove blocks in the pattern instance, this should be fine. @glendaviesnz Do you spot any concerns?

Reply:
Well, nvm. Seems like it's still needed for some reason 😅.